plugins/modules/ec2_vpc_vgw.py

```diff
@@ -270,50 +291,48 @@ def create_vgw(client, module):
         params["AmazonSideAsn"] = module.params.get("asn")
 
     try:
-        response = client.create_vpn_gateway(aws_retry=True, **params)
-        get_waiter(client, "vpn_gateway_exists").wait(VpnGatewayIds=[response["VpnGateway"]["VpnGatewayId"]])
+        response = create_vpn_gateway(client, **params)
+        get_waiter(client, "vpn_gateway_exists").wait(VpnGatewayIds=[response["VpnGatewayId"]])
     except botocore.exceptions.WaiterError as e:
-        module.fail_json_aws(
-            e, msg=f"Failed to wait for Vpn Gateway {response['VpnGateway']['VpnGatewayId']} to be available"
-        )
+        module.fail_json_aws(e, msg=f"Failed to wait for Vpn Gateway {response['VpnGatewayId']} to be available")
     except is_boto3_error_code("VpnGatewayLimitExceeded") as e:
         module.fail_json_aws(e, msg="Too many VPN gateways exist in this account.")
-    except (
-        botocore.exceptions.ClientError,
-        botocore.exceptions.BotoCoreError,
-    ) as e:  # pylint: disable=duplicate-except
-        module.fail_json_aws(e, msg="Failed to create gateway")
+    except AnsibleEC2Error as e:
+        module.fail_json_aws_error(e)
 
     result = response
     return result
 
 
-def delete_vgw(client, module, vpn_gateway_id):
+def delete_vgw(client, module: AnsibleAWSModule, vpn_gateway_id: str) -> Optional[str]:
     try:
-        response = client.delete_vpn_gateway(VpnGatewayId=vpn_gateway_id, aws_retry=True)
-    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-        module.fail_json_aws(e, msg="Failed to delete gateway")
+        delete_vpn_gateway(client, vpn_gateway_id)
+    except AnsibleEC2Error as e:
+        module.fail_json_aws_error(e)
 
     # return the deleted VpnGatewayId as this is not included in the above response
     result = vpn_gateway_id
     return result
 
 
-def find_vpc(client, module):
+def find_vpc(client, module: AnsibleAWSModule) -> Optional[Any]:
     params = dict()
-    params["vpc_id"] = module.params.get("vpc_id")
+    vpc_id = module.params.get("vpc_id")
 
-    if params["vpc_id"]:
+    if vpc_id:
+        params["VpcIds"] = [vpc_id]
         try:
-            response = client.describe_vpcs(VpcIds=[params["vpc_id"]], aws_retry=True)
-        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-            module.fail_json_aws(e, msg="Failed to describe VPC")
+            response = describe_vpcs(client, **params)
+        except AnsibleEC2Error as e:
+            module.fail_json_aws_error(e)
 
     result = response
     return result
 
 
-def find_vgw(client, module, vpn_gateway_id=None):
+def find_vgw(
+    client, module: AnsibleAWSModule, vpn_gateway_id: Optional[Union[str, List[str]]] = None
+) -> List[Dict[str, Any]]:
     params = dict()
     if vpn_gateway_id:
         params["VpnGatewayIds"] = vpn_gateway_id
```

Comment:
```suggestion
        params["VpnGatewayIds"] = [vpn_gateway_id]
```

Reply:
not sure what happens with the API but it fails with
```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: Invalid type for parameter VpnGatewayIds[0], value: ['vgw-0849b49ec813d18ee'], type: <class 'list'>, valid types: <class 'str'>
```

Reply:
Same as below.

Reply:
Hi @alinabuzachis , fixed. It was indeed being passed as List[Str] at the API call.
Only thing was that it was handled to be passed as List[Str] earlier in the function call tree.
Updated the code to ensure it is passed as List[Str] before calling module_utils/ec2 functions.
https://github.com/ansible-collections/community.aws/pull/2171/commits/27b07e056bccfd390b8c1425acfdaa6ac58a44af